src/FFmpegReader.cpp

```diff
@@ -926,11 +927,18 @@ std::shared_ptr<Frame> FFmpegReader::ReadStream(int64_t requested_frame) {
 	bool check_seek = false;
 	int packet_error = -1;
 
+	// Seeking can sometimes cause us to be quite far away from the requested
+	// frame, and thus, we need to iterate a large number of packets to find
+	// the requested position. This is mostly to prevent infinite loops when
+	// no more packets can be found though.
+	int attempts = 0;
+	int max_attempts = 30 * 30 * 60;
+
 	// Debug output
 	ZmqLogger::Instance()->AppendDebugMethod("FFmpegReader::ReadStream", "requested_frame", requested_frame, "max_concurrent_frames", max_concurrent_frames);
 
 	// Loop through the stream until the correct frame is found
-	while (true) {
+	while (attempts < max_attempts) {
```

Comment:
Typically this loop never approaches the max_attempts, and instead breaks out of the loop as soon as the requested frame is found, or as soon as all packets have been decoded, typically within a few iterations. However, there are rare cases where an infinite loop can arise, and we don't want this to loop forever.

Reply:
Mostly caused by manual override to the `has_video` or `has_audio` properties after the reader has called Open(), leaving a few read packets undecoded.